[CI] ServiceUtilsTests testRemoveAsAdaptiveAllocationsSettings_exceptions failing

**Build Scans:**
- [elasticsearch-periodic #3429 / release-tests](https://gradle-enterprise.elastic.co/s/vvr63smyqopsa)
- [elasticsearch-periodic #3397 / release-tests](https://gradle-enterprise.elastic.co/s/43w4iz7qfrdio)
- [elasticsearch-periodic #3393 / release-tests](https://gradle-enterprise.elastic.co/s/ot45vrnwfdzby)
- [elasticsearch-periodic #3389 / release-tests](https://gradle-enterprise.elastic.co/s/q63uzqzat7u5e)
- [elasticsearch-periodic #3385 / release-tests](https://gradle-enterprise.elastic.co/s/hm66mjjfjko4y)
- [elasticsearch-periodic #3381 / release-tests](https://gradle-enterprise.elastic.co/s/l54du3phmdkoo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:test" --tests "org.elasticsearch.xpack.inference.services.ServiceUtilsTests.testRemoveAsAdaptiveAllocationsSettings_exceptions" -Dtests.seed=258786E08226296F -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=ja-JP -Dtests.timezone=Europe/Saratov -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRemoveAsAdaptiveAllocationsSettings_exceptions'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.inference.services.ServiceUtilsTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size <3>
     but: collection size was <0>
```

**Issue Reasons:**
- [main] 6 consecutive failures in step release-tests
- [main] 6 failures in test testRemoveAsAdaptiveAllocationsSettings_exceptions (1.2% fail rate in 511 executions)
- [main] 6 failures in step release-tests (100.0% fail rate in 6 executions)
- [main] 6 failures in pipeline elasticsearch-periodic (46.2% fail rate in 13 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/ServiceUtilsTests.java
+++ b/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/ServiceUtilsTests.java
@@ -21,6 +21,7 @@ import org.elasticsearch.inference.TaskType;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.xpack.core.inference.results.InferenceTextEmbeddingByteResults;
 import org.elasticsearch.xpack.core.inference.results.InferenceTextEmbeddingFloatResults;
+import org.elasticsearch.xpack.core.ml.inference.assignment.AdaptiveAllocationsFeatureFlag;
 import org.elasticsearch.xpack.core.ml.inference.assignment.AdaptiveAllocationsSettings;
 import org.elasticsearch.xpack.inference.results.InferenceTextEmbeddingByteResultsTests;
 import org.elasticsearch.xpack.inference.results.TextEmbeddingResultsTests;
@@ -289,6 +290,8 @@ public class ServiceUtilsTests extends ESTestCase {
     }
 
     public void testRemoveAsAdaptiveAllocationsSettings() {
+        assumeTrue("Should only run if adaptive allocations feature flag is enabled", AdaptiveAllocationsFeatureFlag.isEnabled());
+
         Map<String, Object> map = new HashMap<>(
             Map.of("settings", new HashMap<>(Map.of("enabled", true, "min_number_of_allocations", 7, "max_number_of_allocations", 42)))
         );
@@ -311,6 +314,8 @@ public class ServiceUtilsTests extends ESTestCase {
     }
 
     public void testRemoveAsAdaptiveAllocationsSettings_exceptions() {
+        assumeTrue("Should only run if adaptive allocations feature flag is enabled", AdaptiveAllocationsFeatureFlag.isEnabled());
+
         Map<String, Object> map = new HashMap<>(
             Map.of("settings", new HashMap<>(Map.of("enabled", "YES!", "blah", 42, "max_number_of_allocations", -7)))
         );

